src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/layout/update/updateFormLayoutSagas.ts

```diff
@@ -222,14 +222,29 @@ export function* watchCalculatePageOrderAndMoveToNextPageSaga(): SagaIterator {
 }
 
 export function* watchInitialCalculagePageOrderAndMoveToNextPageSaga(): SagaIterator {
-  yield all([
-    take(START_INITIAL_DATA_TASK_QUEUE_FULFILLED),
-    take(FormLayoutActions.fetchLayoutFulfilled),
-    take(FormLayoutActions.fetchLayoutSettingsFulfilled),
-  ]);
-  const state: IRuntimeState = yield select();
-  if (state.formLayout.uiConfig.layoutOrder.length > 1) {
-    yield put(FormLayoutActions.calculatePageOrderAndMoveToNextPage({ skipMoveToNext: true }));
+  while (true) {
+    yield all([
+      take(START_INITIAL_DATA_TASK_QUEUE_FULFILLED),
+      take(FormLayoutActions.fetchLayoutFulfilled),
+      take(FormLayoutActions.fetchLayoutSettingsFulfilled),
+    ]);
+    const state: IRuntimeState = yield select();
+    const layouts = state.formLayout.layouts;
+    const pageTriggers = state.formLayout.uiConfig.pageTriggers;
+    const appHasCalculateTrigger = pageTriggers?.includes(Triggers.CalculatePageOrder) || Object.keys(layouts).some((layout) => {
```

Comment:
Do we actually need to check all layouts here, or just the "current" layout?

Reply:
I think we need to actually check all pages to avoid the user being able to go back to a page that is supposed to be hidden.

An app with ["page1", "page2", "page3", "page4"]

Users fill out something on page 1 which has a calc trigger and hides page 2, then navigates to page 3. Then closes browser and returns. If we do not have a call to calculate page order on initial reload on page 3, the user could potentially navigate back to page 2.

Reply:
Ah, good point!